src/common/TreeOps.h

```diff
@@ -25,22 +26,21 @@ class TermVisitor {
 public:
     TermVisitor(Logic const & logic, TConfig & cfg) : logic(logic), cfg(cfg) {}
 
-    virtual void visit(PTRef root) {
-        // Avoid initializations if no traversal will be done
-        if (logic.isVar(root)) {
-            if (cfg.previsit(root))
-                cfg.visit(root);
-            return;
-        }
+    void visit(opensmt::span<const PTRef> roots) {
         struct DFSEntry {
-            DFSEntry(PTRef term) : term(term) {}
+            explicit DFSEntry(PTRef term) : term(term) {}
             PTRef term;
             unsigned int nextChild = 0;
         };
```

Comment:
It is local to this function, just an implementation detail. So for me, it logically belongs here.

Reply:
I mean there are no others at the moment, but generally there may be others in the future. It is not important. I just think about it in a way if the class was extended, the `DFSEntry` struct would not have to be moved out of the function. It happened to me many times that I believed a struct would not be necessary anywhere outside of a scope, but in the end, it was.

Reply:
I think this class is a bit specific, because customization happens through the `Config`.

In general, I don't see a problem in making this (and other changes) if the need arises, but I also don't think it makes sense to make premature changes because it "might be potentially useful in the future".